internal/bloblang/query/methods_strings.go

```diff
@@ -1465,22 +1470,19 @@ func replaceAllManyImpl(args *ParsedParams) (simpleMethod, error) {
 		if err != nil {
 			return nil, fmt.Errorf("invalid replacement value at index %v: %w", i+1, err)
 		}
-		replacePairs = append(replacePairs, [2]string{from, to})
-		replacePairsBytes = append(replacePairsBytes, [2][]byte{[]byte(from), []byte(to)})
+		replacePairs[i] = from
+		replacePairs[i+1] = to
 	}
 
+	strReplacer := strings.NewReplacer(replacePairs...)
+	byteReplacer := bytereplacer.New(replacePairs...)
+
 	return func(v any, ctx FunctionContext) (any, error) {
 		switch t := v.(type) {
 		case string:
-			for _, pair := range replacePairs {
-				t = strings.ReplaceAll(t, pair[0], pair[1])
-			}
-			return t, nil
+			return strReplacer.Replace(t), nil
 		case []byte:
-			for _, pair := range replacePairsBytes {
-				t = bytes.ReplaceAll(t, pair[0], pair[1])
-			}
-			return t, nil
+			return byteReplacer.Replace(t), nil
```

Comment:
@Jeffail Should I clone `t` here?  The byte replace is make in-place modifications to `t`.